src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -2661,14 +2658,8 @@ int xclInternalResetDevice(xclDeviceHandle handle, xclResetKind kind)
 
 int xclP2pEnable(xclDeviceHandle handle, bool enable, bool force)
 {
-  try {
     xocl::shim *drv = xocl::shim::handleCheck(handle);
     return drv ? drv->p2pEnable(enable, force) : -ENODEV;
-  }
-  catch (const std::exception& ex) {
-    xrt_core::send_exception_message(ex.what());
-    return -ENODEV;
-  }
```

Comment:
No. this will not work.  This can be called from host C application and now it propagates an exception.